src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/FlowScanningUtils.java

```diff
@@ -59,6 +62,45 @@ public boolean apply(FlowNode input) {
     // Default predicates, which may be used for common conditions
     public static final Predicate<FlowNode> MATCH_BLOCK_START = (Predicate)Predicates.instanceOf(BlockStartNode.class);
 
+    /** Sorts flownodes putting the one begun last (oldest startTime) at the end, with null times last
+     *  because likely they represent the newest nodes with a {@link TimingAction} not attached yet. */
+    public static final Comparator<FlowNode> TIME_ORDER_COMPARATOR = new Comparator<FlowNode>() {
```

Comment:
@rsandell If we'd only had Java 8 functional features to play with  this would use 'em and the streaming APIs ;)